tests/test_writers.py

```diff
@@ -0,0 +1,406 @@
+from unittest import TestCase
```

Comment:
Might be nice to break this up into `tests/transaction_writers/detail.py` and so on.